src/wasm-interpreter.h

```diff
@@ -149,11 +149,51 @@ class Indenter {
 // Execute an expression
 template<typename SubType>
 class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
+public:
+  enum FlagValues {
+    // By default, just evaluate the expression, i.e. all we want to know is
+    // whether it computes down to a concrete value, where it is not necessary
+    // to preserve side effects like those of a `local.tee`.
+    DEFAULT = 0,
+    // Be very careful to preserve any side effects, like those of a
+    // `local.tee`, for example when we are going to replace the expression
+    // afterwards.
+    PRESERVE_SIDEEFFECTS = 1 << 0,
+    // Traverse through function calls, attempting to compute their concrete
+    // value. Must not be used in function-parallel scenarios, where the called
+    // function might or might not have been optimized already to something we
+    // can traverse successfully, in turn leading to non-deterministic behavior.
+    TRAVERSE_CALLS = 1 << 1
+  };
+
+  // Flags indicating special requirements, for example whether we are just
+  // evaluating (default), also going to replace the expression afterwards or
+  // executing in a function-parallel scenario. See FlagValues.
+  typedef uint32_t Flags;
+
 protected:
-  Index maxDepth;
+  // Flags indicating special requirements. See FlagValues.
+  Flags flags;
 
+  // Maximum depth before giving up. Defaults to 0 for no limit.
```

Comment:
It's risky to put stuff like "defaults to 0" in comments, as they tend to become stale.

I think it's best to just assign to 0 here in code, and the same later down.